handlers.go

```diff
@@ -70,9 +70,27 @@ func receiveHandler(producer *kafka.Producer, serializer Serializer) func(c *gin
 			if err != nil {
 				c.AbortWithStatus(http.StatusInternalServerError)
 				logrus.WithError(err).Error("couldn't produce message in kafka")
+				metricsHandledTotal.Add(float64(i))
 				return
 			}
 		}
+		metricsHandledTotal.Add(float64(len(metrics)))
+	}
+}
 
+// kafkaErrorsHandler reads events off of a producer's event channel, logging
+// errors and exposing counters to prometheus by error code. Always call as
+// `go kafkaErrorsHandler(...)` as this function never returns
+func kafkaErrorsHandler(events <-chan (kafka.Event)) {
+	for {
+		select {
+		case evt := <-events:
+			switch e := evt.(type) {
+			case kafka.Error:
+				logrus.WithError(e).Errorf("Error returned in Kafka event")
+				kafkaErrors.WithLabelValues(e.Code().String())
+			default:
```

Comment:
should we log that this is unexpected?

Reply:
example code doesn't have a default case
```
	switch ev := e.(type) {
			case *kafka.Message:
				if ev.TopicPartition.Error != nil {
					fmt.Printf("Delivery failed: %v\n", ev.TopicPartition)
				} else {
					fmt.Printf("Delivered message to %v\n", ev.TopicPartition)
				}
			}
```

Reply:
I based this on somebody's article: https://medium.com/@abhishek1987/kafka-go-client-no-producer-error-during-connection-b3d4116b91ff, which seems to be a slightly different checking format than that. I wonder if they're picking up different types of errors, or they're functionally equivalent. I just had a default case out of habit, and then didn't delete it. I'll remove it since it doesn't actually add anything